modules/pam_motd/pam_motd.c

```diff
@@ -215,6 +209,45 @@ static void try_to_display_directories_with_overrides(pam_handle_t *pamh,
     if (dirscans_size_total == 0)
         goto out;
 
+    /* filter out unwanted names, directories, and complement data with lstat() */
+    for (i = 0; i < num_motd_dirs; i++) {
+        unsigned int j;
+        int rc, drop;
+        struct dirent **d;
+	char *fullpath;
+	struct stat s;
+
+        d=dirscans[i];
+        for ( j=0; j<dirscans_sizes[i]; j++ ) {
+            drop=1;
+            switch( d[j]->d_type ) {  /* the filetype determines how to proceed */
+            case DT_REG:              /* regular files and     */
+            case DT_LNK:              /* symlinks              */
+		drop=0;               /* are good.             */
+		break;
+            case DT_UNKNOWN:   /* for file systems that do not provide */
+			       /* a filetype, we use lstat()           */
+	        if ( join_dir_strings( &fullpath, motd_dir_path_split[i],
+				       d[j]->d_name )<=0 ) break;
+                rc=lstat( fullpath, &s );
+                _pam_drop(fullpath);  /* free the memory alloc'ed by join_dir_strings */
+                if ( rc!=0 ) break;   /* if the lstat() somehow failed */
```

Comment:
Please use consistent indentation.

Please follow the style used in the file being patched, e.g.
`=`, `!=`, and `<=` should be surrounded by spaces,
the should be no space after `(` and before `)`,
there should be a space after `switch` keyword,
`break;` statement should be placed on a separate line.

Reply:
Understood.